### ISSUE TITLE
Connecting to Self Hosted Instances of Canvas on Private Server

### ISSUE THREAD

**Issue by gee455** [instructure/instructure-ios#17]
Hi!

I'm struggling to figure out where to modify the code in order to properly link the app to my self hosted instance of Canvas hosted on a private server. 

Any help is appreciated! Thanks!

**Comment by gee455** [comment]
Hey! Thanks so much for the reply @laynemoseley. I don't think that's exactly what I'm looking for. 

Here's an imgur gallery of the domains I'm talking about: https://imgur.com/a/QOfotO3. 

Basically, instead of hooking up to Canvas' official servers at like learn.instructure.net or [organization/school].instructure.com. I'd like to point it to my private server at MyOrganization.com.

So I'm trying to see which files I would need to change to get that linked up.

Thanks again!

**Comment by laynemoseley** [comment]
You have an instance of canvas running in the web, right?

**Comment by gee455** [comment]
Correct :). Currently at - https://blitzprepuniversity.com/

**Comment by laynemoseley** [comment]
so the only way to do this currently is with `preload-account-info.plist`. We don't have an official supported way to do what you're looking for currently.

**Comment by gee455** [comment]
Ah ok so if I modify that .plist file with my URL, I should be able to login to my instance of canvas?

**Comment by laynemoseley** [comment]
That's correct. You'll need to supply all of the information in that plist.

**Comment by gee455** [comment]
Thank you so much, @laynemoseley!!! Really really appreciate the help!

**Comment by vdharankar** [comment]
@laynemoseley tried your suggestion of changing the plist file mentioned above , but real issue after that is oAuth 2 redirect_url setting , we ended up with {"error":"invalid_request",:error_description":"redirect_uri does not match client settings"}
when we clicked on custom login button ( with our domain name label at the top of the main app screen)  .

we have redirect uri set as https://mydomain.com/oauth_completeoauth_complete.
not sure why still getting above error.

**Comment by shichongrui** [comment]
Currently to make this work you need to set your redirect uri for your developer key in canvas to `https://canvas/login`. Canvas will inform the app to redirect to that uri and then the app will know how to complete the oauth flow from there.
